src/commands.ts

```diff
@@ -1132,89 +1004,105 @@ const getParentNode = (selection: Selection) => {
   state: ManuscriptEditorState,
   dispatch?: Dispatch
 ) => {
-  const {
-    attrs: { id },
-    type,
-  } = getNode(node)
-  const comment = buildComment(id)
-  const tr = state.tr.setMeta(commentAnnotation, {
-    [SET_COMMENT]: comment,
-  })
-
-  tr.setMeta('addToHistory', false)
-
-  if (isAllowedType(type)) {
-    dispatch && dispatch(tr)
-    return true
-  } else {
+  if (!isCommentingAllowed(node.type)) {
     return false
   }
-}
 
-const addHighlightComment = (
-  node: ManuscriptNode,
-  state: ManuscriptEditorState,
-  dispatch?: Dispatch
-) => {
-  const {
-    attrs: { id },
-    type,
-  } = node
-  const comment = buildComment(id) as CommentAnnotation
-
-  if (type === schema.nodes.paragraph) {
-    const { $anchor, $head } = state.selection
-    const isAllTextSelected =
-      ($anchor.textOffset === 0 && $head.textOffset === 0) ||
-      $anchor.textOffset === $head.textOffset
-
-    if (isAllTextSelected) {
-      return addBlockComment(node, state, dispatch)
-    } else {
-      return insertHighlight(state, comment, dispatch)
-    }
-  } else {
-    return insertHighlight(state, comment, dispatch)
+  const props = getEditorProps(state)
+  const contribution = buildContribution(props.userID)
+  const attrs = {
+    id: generateID(ObjectTypes.CommentAnnotation),
+    contents: '',
+    target: node.attrs.id,
+    contributions: [contribution],
+  } as CommentAttrs
+  const comment = schema.nodes.comment.create(attrs)
+  const comments = findChildrenByType(state.doc, schema.nodes.comments)[0]
+  const pos = comments.pos + 1
+
+  const tr = state.tr.insert(pos, comment)
+  const key = getCommentKey(attrs, undefined, node)
+  setCommentSelection(tr, key, attrs.id, true)
+  if (dispatch) {
+    dispatch(tr)
   }
+  return true
 }
 
-export const updateCommentAnnotationState = (
+export const addInlineComment = (
   state: ManuscriptEditorState,
   dispatch?: Dispatch
-) => {
-  const tr = state.tr.setMeta(commentAnnotation, {})
-
-  tr.setMeta('addToHistory', false)
-
+): boolean => {
+  const selection = state.selection
+  const node = getParentNode(selection)
+  if (!node) {
+    return false
+  }
+  const from = selection.from
+  const to = selection.to
+
+  const props = getEditorProps(state)
+  const contribution = buildContribution(props.userID)
+  const attrs = {
+    id: generateID(ObjectTypes.CommentAnnotation),
+    contents: '',
+    target: node.attrs.id,
+    contributions: [contribution],
+    originalText: selectedText(),
+    selector: {
+      from,
+      to,
+    },
+  } as CommentAttrs
+  const comment = schema.nodes.comment.create(attrs)
+  const comments = findChildrenByType(state.doc, schema.nodes.comments)[0]
+  const pos = comments.pos + 1
+
+  const tr = state.tr.insert(pos, comment)
+
+  if (from === to) {
+    const point = schema.nodes.highlight_marker.create({
+      id: comment.attrs.id,
+      tid: node.attrs.id,
+      position: 'point',
+    })
+    tr.insert(from, point)
```

Comment:
so now we support adding point comments? as it was expected in that case to add block comment. and I think that will be problematic to do for any node not just text

Reply:
Is there a specific case where this will be a problem? this is supposed to be for adding "inline" comments so it should not be called for non-text nodes.

Reply:
![Screenshot from 2024-06-05 19-03-35](https://github.com/Atypon-OpenSource/manuscripts-body-editor/assets/45534101/f6be0a14-9154-45fb-be29-0a02e9b1a666)
![Screenshot from 2024-06-05 19-03-41](https://github.com/Atypon-OpenSource/manuscripts-body-editor/assets/45534101/7005e127-e47b-48c1-b1ce-6fcee56995cd)

Reply:
I updated the code to check whether the target node is a valid target. This should fix the issue,